From 6629392d8b44ddfacaaca0ae8afc7f61ae9d3412 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:51 -0400
Subject: [usb] ssb-hcd: replace dma_set_mask()+dma_set_coherent_mask() with new helper

Message-id: <1410887463-20674-160-git-send-email-dzickus@redhat.com>
Patchwork-id: 95404
O-Subject: [RHEL7 PATCH 159/591] DMA-API: usb: ssb-hcd: replace dma_set_mask()+dma_set_coherent_mask() with new helper
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 01edb5e12c7fd2170b1969a455442e8fd17cfe54
Author: Russell King <rmk+kernel@arm.linux.org.uk>
Date:   Wed Jun 26 23:49:11 2013 +0100

    DMA-API: usb: ssb-hcd: replace dma_set_mask()+dma_set_coherent_mask() with new helper

    Replace the following sequence:

     dma_set_mask(dev, mask);
     dma_set_coherent_mask(dev, mask);

    with a call to the new helper dma_set_mask_and_coherent().

    Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ssb-hcd.c b/drivers/usb/host/ssb-hcd.c
index 74af2c6..0196f76 100644
--- a/drivers/usb/host/ssb-hcd.c
+++ b/drivers/usb/host/ssb-hcd.c
@@ -163,8 +163,7 @@ static int ssb_hcd_probe(struct ssb_device *dev,
 
  /* TODO: Probably need checks here; is the core connected? */
 
- if (dma_set_mask(dev->dma_dev, DMA_BIT_MASK(32)) ||
-     dma_set_coherent_mask(dev->dma_dev, DMA_BIT_MASK(32)))
+ if (dma_set_mask_and_coherent(dev->dma_dev, DMA_BIT_MASK(32)))
   return -EOPNOTSUPP;
 
  usb_dev = kzalloc(sizeof(struct ssb_hcd_device), GFP_KERNEL);
-- 
1.7.1